### PR TITLE
[zh-cn]: Update translation of Array.find.

### PR DESCRIPTION
**Pull request by JasonLamv-t**
### Description

更新了 `Array.find()`

### Motivation

同上

### Additional details

🈚️

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/global_objects/array/find/index.md [hunk 1]
```diff
@@ -7,79 +7,85 @@ slug: Web/JavaScript/Reference/Global_Objects/Array/find
...
     - `array`
-      - : 数组本身。
-
-    回调函数必须返回一个[真值](/zh-CN/docs/Glossary/Truthy)来表示找到了匹配的元素。
+      - : 调用 `find()` 的当前数组。
 
-- `thisArg` {{Optional_inline}}
-  - : 执行回调时用作 `this` 的对象。
+- `thisArg` {{optional_inline}}
+  - : 执行 `callbackFn` 时使用的 `this` 值。请参阅[迭代方法](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array#iterative_methods)。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/find/index.md:52]:
```suggestion
  - : 执行 `callbackFn` 时使用的 `this` 值。请参阅[迭代方法](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array#迭代方法)。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/find/index.md [hunk 2]
```diff
@@ -90,14 +96,14 @@ console.log(inventory.find(isCherries));
...
 ];
 
-const result = inventory.find(({ name }) => name === 'cherries');
+const result = inventory.find(({ name }) => name === "cherries");
 
-console.log(result) // { name: 'cherries', quantity: 5 }
+console.log(result); // { name: 'cherries', quantity: 5 }
 ```
 
 ### 寻找数组中的第一个质数
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/find/index.md:109]:
```suggestion
### 寻找数组中的第一个素数
```

**Review by JasonLamv-t — COMMENTED** [review]
WIP

File: files/zh-cn/web/javascript/reference/global_objects/array/find/index.md [hunk 3]
```diff
@@ -7,79 +7,85 @@ slug: Web/JavaScript/Reference/Global_Objects/Array/find
...
 ## 描述
 
-`find` 方法对数组中的每一项元素执行一次 `callbackFn` 函数，直至有一个 callbackFn 返回 `true`。当找到了这样一个元素后，该方法会立即返回这个元素的值，否则返回 {{jsxref("undefined")}}。注意 `callbackFn` 函数会为数组中的每个索引调用即从 `0` 到 `length - 1`，而不仅仅是那些被赋值的索引，这意味着对于稀疏数组来说，该方法的效率要低于那些只遍历有值的索引的方法。
+`find()` 方法是一个[迭代方法](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array#迭代方法)。它按索引升序顺序为数组中的每个元素调用提供的 `callbackFn` 函数，直到 `callbackFn` 返回一个[真值](/zh-CN/docs/Glossary/Truthy)。然后 `find()` 返回该元素并停止迭代数组。如果 `callbackFn` 从未返回真值，则 `find()` 返回 {{jsxref("undefined")}}。
 
-`callbackFn` 函数带有 3 个参数：当前元素的值、当前元素的索引，以及数组本身。
+`callbackFn` 被调用来处理数组的*每一个*索引，而不仅仅是那些有值的索引。在[稀疏数组](/zh-CN/docs/Web/JavaScript/Guide/Indexed_collections#稀疏数组)中，未赋值的空槽与 `undefined` 表现相同。
 
-如果提供了 `thisArg` 参数，那么它将作为每次 `callbackFn` 函数执行时的 `this`，如果未提供，则使用 {{jsxref("undefined")}}。
+`find()` does not mutate the array on which it is called, but the function provided as `callbackFn` can. Note, however, that the length of the array is saved _before_ the first invocation of `callbackFn`. Therefore:
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/array/find/index.md:64]:
```suggestion
```

File: files/zh-cn/web/javascript/reference/global_objects/array/find/index.md [hunk 4]
```diff
@@ -7,79 +7,85 @@ slug: Web/JavaScript/Reference/Global_Objects/Array/find
...
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/array/find/index.md:65]:
```suggestion
```

File: files/zh-cn/web/javascript/reference/global_objects/array/find/index.md [hunk 5]
```diff
@@ -115,47 +121,63 @@ function isPrime(element, index, array) {
   return element > 1;
 }
 
-console.log([4, 6, 8, 12].find(isPrime)); // undefined, not found
+console.log([4, 6, 8, 12].find(isPrime)); // undefined, 未找到
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/array/find/index.md:124]:
```suggestion
console.log([4, 6, 8, 12].find(isPrime)); // undefined，未找到
```

**Review by yin1999 — CHANGES_REQUESTED** [review]

**Review by yin1999 — APPROVED** [review]

**Pull request merged**